docs: Conceptual Docs for Api Gateways, The Gateway Api & Envoy Proxy (#5739)
envoyproxy/gateway: site/content/en/latest/concepts/introduction/api_gateways.md

---
title: "API Gateways"
---

## Overview

> An API gateway is a centralized entry point for managing, securing, and routing requests to backend services. It handles cross-cutting concerns—like authentication, rate limiting, and protocol translation—so individual services don’t have to. By decoupling clients from internal systems, it simplifies scaling, enforces consistency, and reduces redundancy.

Reviewer: ```suggestion
An API gateway is a centralized entry point for managing, securing, and routing requests to backend services. It handles cross-cutting concerns, like authentication, rate limiting, and protocol translation, so individual services don’t have to. Decoupling clients from internal systems simplifies scaling, enforces consistency, and reduces redundancy.
```

## Use Cases

Use an API Gateway to:
- Avoid duplicating logic across microservices.
- Create a central point of control for access, monitoring, and traffic rules.
- Expose internal services to the public internet.
- Provide protocol support for HTTP, gRPC, or TLS.
- Enforce policies and see traffic metrics at the edge.

## API Gateways in Envoy Gateway

> Under the hood, Envoy Proxy is a powerful, production-grade proxy that supports many of the capabilities you'd expect from an API Gateway — like traffic routing, retries, TLS termination, observability, and more. But configuring Envoy directly can be complex and verbose.

Reviewer: ```suggestion
Under the hood, Envoy Proxy is a powerful, production-grade proxy that supports many of the capabilities you'd expect from an API Gateway, like traffic routing, retries, TLS termination, observability, and more. However, configuring Envoy directly can be complex and verbose.
```

## Use Cases

Use an API Gateway to:
- Avoid duplicating logic across microservices.
- Create a central point of control for access, monitoring, and traffic rules.
- Expose internal services to the public internet.
- Provide protocol support for HTTP, gRPC, or TLS.
- Enforce policies and see traffic metrics at the edge.

## API Gateway's in Envoy Gateway

Under the hood, Envoy Proxy is a powerful, production-grade proxy that supports many of the capabilities you'd expect from an API Gateway — like traffic routing, retries, TLS termination, observability, and more. But configuring Envoy directly can be complex and verbose.

Envoy Gateway solves this by providing a Kubernetes-native interface, using the standard Gateway API and its own custom resources, to make configuring Envoy Proxy simpler and more approachable. You define high-level traffic rules using resources like Gateway, HTTPRoute, or TLSRoute, and Envoy Gateway automatically translates them into detailed Envoy Proxy configurations.

## Related Resources

> - [Getting Started with Envoy Gateway](https://gateway.envoyproxy.io/docs/latest/getting-started/)

Reviewer: make this link relative

---
title: "API Gateways"
---

## Overview

> An API Gateway is a centralized entry point that exposes, secures, and orchestrates access to your application’s microservices or backend APIs—it validates and transforms requests, enforces policies, and routes traffic to the appropriate services.

Reviewer: ```suggestion
An API Gateway is a centralized entry point that exposes, secures, and orchestrates access to your application’s microservices or backend APIs. It can validate and transforms requests, enforces policies, and routes traffic to the appropriate upstream services.
```

> In modern cloud-native systems, especially those built on microservices, an API Gateway helps simplify architecture, improve security, and streamline traffic management. Instead of each service handling these concerns individually, the API Gateway offloads and centralizes them, allowing your services to focus on core business logic.

Reviewer: ```suggestion
In cloud-native systems, especially those that expose APIs to applications, an API Gateway helps simplify the architecture, improve security, and streamline traffic management. Instead of each service handling these concerns individually, the API Gateway offloads and centralizes them, allowing your services to focus on core business logic.
```

> ### Understanding Ingress

Reviewer: Consider changing this section to be about Gateway vs Ingress and how Gateway is the evolution of Ingress. Maybe explain how someone coming from a background of using Kubernetes ingress can use and understand Gateway API

Author: Added a section like this in API Gateways and the Gateway API doc!

- Provide protocol support for HTTP, gRPC, or TLS.
- Enforce policies and see traffic metrics at the edge.

> ## API Gateways in Envoy Gateway

Reviewer: ```suggestion
## API Gateways in relation to Envoy Gateway
```
